Source/Transforms/LMCWindowFunctions.cc

```diff
@@ -0,0 +1,172 @@
+/*
+ * LMCWindowFunctions.cc
+ *
+ *  Created on: Jan 10, 2022
+ *      Author: A. B. Telles
+ */
+#include <iostream>
+#include <fstream>
+
+#include "logger.hh"
+#include "LMCWindowFunctions.hh"
+#include "LMCConst.hh"
+
+namespace locust
+{
+    LOGGER( lmclog, "WindowFunctions" );
+
+    WindowFunctions::WindowFunctions():
+    IsWindowGeneratedFlag( false ),
+    fTotalWindowSize(0),
+    fPreWindowBins(0),
+    fSignalSize(0),
+    fWindowFunctionType(0),
+    fWindowFunction( 1 ),
+    fTukeyWindowAlpha( 0.5 )
+    {
+    }
+    
+    WindowFunctions::~WindowFunctions()
+    {
+    }
+    
+    bool WindowFunctions::Configure(const scarab::param_node& aParam)
+    {
+        return true;
+    }
+
+    bool WindowFunctions::SetupWindow(std::string windowname, double windowparam)
+    {
+        if(windowname == "rectangular")
+        {
+            fWindowFunctionType = 0;
+        }
+        else if(windowname == "tukey")
+        {
+            fWindowFunctionType = 1;
+            fTukeyWindowAlpha = windowparam;
+        }
+        else if(windowname == "hanning")
+        {
+            fWindowFunctionType = 2;
+        }
+        else if(windowname == "hann")
+        {
+            fWindowFunctionType = 2;
+        }
+        else
+        {
+            return false;
+        }
+        return true;
+    }
+    
+    bool WindowFunctions::GenerateWindowFunction(int windowsize, int prewindowbins, int signalsize)
+    {
+        fTotalWindowSize = windowsize;
+        fPreWindowBins = prewindowbins;
+        fSignalSize = signalsize;
+
+        for (int i = 0; i < fTotalWindowSize; ++i)
+        {
+            fWindowFunction.push_back(0.0);
+        }
+
+        if(fWindowFunctionType==0)
+        {
+            GenerateRectangularWindow();
+	    }
+        else if(fWindowFunctionType==1)
+        {
+            GenerateTukeyWindow();
+        }
+        else if(fWindowFunctionType==2)
+        {
+            GenerateHanningWindow();
+        }	
+        else
+        {
+            return false;
+        }
+
+        fWindowPtr = &fWindowFunction;
```

Comment:
Could remove this line if you are returning the reference to `fWindowFunction` as suggested